packages/scene-composer/src/components/panels/CustomColorPicker/CustomColorPicker.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useRef, useState } from 'react';
+import { HexColorPicker } from 'react-colorful';
+import { debounce } from 'lodash';
+import { Button, FormField, InputProps, SpaceBetween } from '@awsui/components-react';
+import './CustomColorPicker.scss';
+import { useIntl } from 'react-intl';
+
+import {
+  ColorPickerDivider,
+  ColorPickerWrapper,
+  ColorPreview,
+  FlexibleDiv,
+  RGB,
+  RgbToHex,
+  hexToRgb,
+  isValidHex,
+  validateHex,
+  validateRgbValue,
+  FixedWidthInput,
+} from './CustomColorPickerHelpers';
+
+export const DEFAULT_COLOR_PICKER_COLOR = {
+  hex: '#ffffff',
+  rgb: {
+    r: '255',
+    g: '255',
+    b: '255',
+  },
+};
+
+interface CustomColorPickerProps {
+  color: string | undefined;
+  onSubmit: (newColor: string) => void;
+  onCancel?: () => void;
+}
+
+export const CustomColorPicker: React.FC<CustomColorPickerProps> = ({
+  color,
+  onSubmit,
+  onCancel,
+}: CustomColorPickerProps) => {
+  const { formatMessage } = useIntl();
+  const hexInputFieldRef = useRef<InputProps.Ref>(null);
+  const [saveAttempted, setSaveAttempted] = useState(false);
+  const [internalColor, setInternalColor] = useState<string>(() =>
+    !!color && isValidHex(color) ? color : DEFAULT_COLOR_PICKER_COLOR.hex,
+  );
+  const [hexInputValue, setHexInputValue] = useState<string>(() =>
+    !!color && isValidHex(color) ? color : DEFAULT_COLOR_PICKER_COLOR.hex,
+  );
+  const [rgbInputValue, setRgbInputValue] = useState<RGB>(() => {
+    return !!color && isValidHex(color) ? hexToRgb(color) : DEFAULT_COLOR_PICKER_COLOR.rgb;
+  });
+
+  const handleColorfulChange = (newColor: string) => {
+    setInternalColor(newColor);
+    setHexInputValue(newColor);
+    setRgbInputValue(hexToRgb(newColor));
+  };
+
+  const handleHexChange = (newColor: string) => {
+    setHexInputValue(validateHex(newColor));
+    if (isValidHex(newColor)) {
+      setInternalColor(newColor);
+      setRgbInputValue(hexToRgb(newColor));
+    }
+  };
+
+  const handleRGBChange = (newValue: Partial<RGB>) => {
+    setRgbInputValue({ ...rgbInputValue, ...newValue });
+    setHexInputValue(RgbToHex({ ...rgbInputValue, ...newValue }));
+    setInternalColor(RgbToHex({ ...rgbInputValue, ...newValue }));
+  };
+
+  const onSubmitAttempt = () => {
+    setSaveAttempted(true);
+    //The hex input is the only field that could conceivably be invalid
+    const valid = isValidHex(hexInputValue);
+    if (valid) {
+      onSubmit(internalColor);
+    } else {
+      hexInputFieldRef.current?.focus();
+    }
```

Comment:
Thoughts on handling errors during user input (in handleHexChange)? Kind of like how Polaris tag editor handles errors immediately during user input. I know for forms we often wait until submission, but for something smaller like this, it might not be bad to show they have a bad hex immediately

Reply:
Yeah that's a good point. Like you said, polaris team recommends waiting until after a first submission until displaying errors which is why I did it this way, but for a small form like this it might be better to validate every change. I am planning to ask Brenton about the save/cancel flow anyway, I'll see what he thinks about this as well.